File: internal/model/Job.go
package model

// JobStatus 是打印任务状态。
type JobStatus string

const (
	JobQueued   JobStatus = "queued"
	JobPrinting JobStatus = "printing"
	JobWaiting  JobStatus = "waiting" // 网络异常,等待重试(堆在任务列表,网络恢复后自动打出)
	JobDone     JobStatus = "done"
	JobFailed   JobStatus = "failed"
)

// Label 返回任务状态的中文标签。
func (s JobStatus) Label() string {
	switch s {
	case JobQueued:
		return "排队中"
	case JobPrinting:
		return "打印中"
	case JobWaiting:
		return "等待重试"
	case JobDone:
		return "已完成"
	case JobFailed:
		return "失败"
	default:
		return string(s)
	}
}

// Active 表示任务是否仍在进行(排队 / 打印中 / 等待重试)。
func (s JobStatus) Active() bool {
	return s == JobQueued || s == JobPrinting || s == JobWaiting
}

// Job 是一次打印任务(用于展示)。
type Job struct {
	No        int
	Doc       string
	PrinterID string
	Printer   string
	Copies    int
	Status    JobStatus
	Time      string
	Err       string
}

File: Android/app/src/main/java/com/congmingpay/android/model/Job.kt
package com.congmingpay.android.model

/**
 * 打印任务状态。
 */
object JobStatus {
    const val QUEUED = "queued"
    const val PRINTING = "printing"
    const val WAITING = "waiting"   // 网络异常，等待重试
    const val DONE = "done"
    const val FAILED = "failed"

    fun label(status: String): String = when (status) {
        QUEUED -> "排队中"
        PRINTING -> "打印中"
        WAITING -> "等待重试"
        DONE -> "已完成"
        FAILED -> "失败"
        else -> status
    }

    /** 任务是否仍在进行（排队 / 打印中 / 等待重试） */
    fun active(status: String): Boolean =
        status == QUEUED || status == PRINTING || status == WAITING
}

/**
 * 打印任务（展示用）。多份打印（pCopy）= 多个任务，每个 Job 恒为一份。
 */
data class Job(
    val no: Int,
    val doc: String,
    val printer: String,
    val status: String,
    val time: String,
    val err: String = ""
)
